src/shared/src/impls.rs

```diff
@@ -45,3 +48,34 @@ impl TokenVersion for CustomToken {
         cloned
     }
 }
+
+impl StoredUserProfile {
+    #[must_use]
+    pub fn default(now: u64) -> StoredUserProfile {
+        let credentials: BTreeMap<CredentialType, UserCredential> = BTreeMap::new();
+        StoredUserProfile {
+            credentials,
+            created_timestamp: now,
+            updated_timestamp: now,
+            version: None,
+        }
+    }
+}
+
+impl UserProfile {
+    #[must_use]
+    pub fn from_stored(user: StoredUserProfile) -> UserProfile {
```

Comment:
That's the pattern we have for tokens

```
impl From<&Token> for CustomTokenId {
    fn from(token: &Token) -> Self {
```

For consistency, should we replicate it here?

Reply:
Done thanks!